crates/eth-rpc/tests/rpc.rs

```diff
@@ -1,20 +1,93 @@
 #![recursion_limit = "1024"]
+mod kkrt_prep;
 mod utils;
 
 #[cfg(test)]
 mod tests {
     use std::str::FromStr;
 
+    use dojo_test_utils::sequencer::TestSequencer;
     use kakarot_rpc::eth_api::EthApiServer;
+    use kakarot_rpc::eth_rpc::KakarotEthRpc;
+    use kakarot_rpc_core::client::config::StarknetConfig;
+    use kakarot_rpc_core::client::KakarotClient;
     use kakarot_rpc_core::mock::assert_helpers::{assert_block, assert_block_header, assert_transaction};
-    use reth_primitives::{BlockNumberOrTag, H160, H256, U256, U64};
+    use reth_primitives::hex_literal::hex;
+    use reth_primitives::{Address, BlockNumberOrTag, H160, H256, U256, U64};
     use reth_rpc_types::Index;
     use serde_json::json;
     use starknet::core::types::{FieldElement, Transaction as StarknetTransaction};
     use starknet::macros::felt;
 
+    use crate::kkrt_prep::{create_call_request, create_raw_tx, init_kkrt_state};
     use crate::utils::setup_kakarot_eth_rpc;
 
+    #[tokio::test]
+    #[ignore]
+    async fn use_test_sequencer_test() {
+        let sn_test_sequencer = TestSequencer::start().await;
+        // TODO: lift from env
+        let kkrt_eoa_address = "foo";
+        let kkrt_eoa_private = H256::from_slice(&hex::decode("bar").unwrap());
+
+        let expected_funded_amount = FieldElement::from_dec_str("100000000000000000").unwrap();
+
+        let (kkrt_address, proxy_account_class_hash, _sn_eoa_address, deployed_addresses) = init_kkrt_state(
+            &sn_test_sequencer,
+            kkrt_eoa_address,
+            kkrt_eoa_private,
+            expected_funded_amount,
+            "Counter.json",
+        )
+        .await;
+
+        let kakarot_client = KakarotClient::new(StarknetConfig::new(
+            sn_test_sequencer.url().as_ref(),
+            kkrt_address,
+            proxy_account_class_hash,
+        ))
+        .unwrap();
+
+        let kkrt_eth_rpc = KakarotEthRpc::new(Box::new(kakarot_client));
+
+        let deployed_balance = kkrt_eth_rpc
+            .balance(
+                Address::from_slice(&hex::decode(kkrt_eoa_address.trim_start_matches("0x")).unwrap()),
+                Option::None,
+            )
+            .await;
+
+        // TODO: clean up how to compare this
+        let deployed_balance = FieldElement::from_bytes_be(&deployed_balance.unwrap().to_be_bytes()).unwrap();
+
+        assert_eq!(deployed_balance, expected_funded_amount);
+
+        let counter_eth_addr = {
+            // TODO  use try_into here when merged
+            let address = &deployed_addresses.first().unwrap().to_bytes_be()[12..];
+            Address::from_slice(address)
+        };
+
+        // asserts there is code at this address
+        // let code_res = kkrt_eth_rpc.get_code(counter_eth_addr, Option::None).await;
+
+        let inc_selector = hex!("371303c0");
```

Comment:
i sourced the selectors from the compiled json, need to look for the equivalent `get_selector_from_name` helper in reth